frontend/src/components/Dish/DishItem.jsx

```diff
@@ -0,0 +1,158 @@
+import React, { useState, useEffect, useContext } from "react";
+import userContext from "../../contexts/context";
+
+const _ = require("lodash");
+
+const apiUtil = require("../../util/apiUtil");
+
+function DishItem(props) {
+  
+    const [dishState, setDish] = useState(() => ({
+      description: {
+        hebrew: "",
+        english: "",
+      },
+      name: {
+        hebrew: "",
+        english: "",
+      },
+      image: {
+        fileLink: "",
+        s3_key: ""
+      },
+      price: 0,
+      businessId: "",
+      isHidden: true,
+    }));
+    
+    const { formType, dish, setFormInfo, isAdmin, setFormCallback } = props;
+    const setUrl = useContext(userContext).setUrl;
+    
+    useEffect(() => {
+      if (formType === "edit") {
+        setDish(() => {
+          const { description, name, image, isHidden, price, businessId } = dish;
+          
+          return {
+            description, name,  image, price, businessId, isHidden
+          };
+        })
+      }
+    }, []);
+  
+    function createFormFields() {
+      let fields = [
+
```

Comment:
Please do auto formatting & indentation, it's really hard to review with all the blank lines